### PR TITLE
chore: upgrade react apollo

### PR DESCRIPTION
**Pull request by craigbilner**
This PR no longer upgrades react apollo but does address mocking issues that were found as part of the upgrade. Need to wait until coming out of beta

### CODE REVIEW

File: packages/provider/connect.js [hunk 1]
```diff
@@ -30,16 +30,13 @@ export const makeGraphqlOptions = (
...
     children,
     ...props
   }) =>
     children({
       error,
-      refetch: () => {
-        retry(); // FIXME: remove this after react-apollo fixes https://github.com/apollographql/apollo-client/issues/2513
-        refetch();
-      },
+      refetch: () => refetch(),
```
Comment (nikhedonia) [inline packages/provider/connect.js]: why not just use a shorthand?

Comment (craigbilner) [reply packages/provider/connect.js]: because the linter didn't telll me off ;-)

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/76c8cb868428c5d59fdaff5bfb937c6716d9e726/dextrose-test/index.html

File: packages/provider/__tests__/__snapshots__/provider-state-transition.test.js.snap [hunk 2]
```diff
@@ -103,6 +103,109 @@ Array [
...
+    "subscribeToMore": [Function],
+    "updateQuery": [Function],
+    "variables": Object {
+      "slug": "1",
+    },
+  },
+  Object {
+    "author": Object {
+      "__typename": "Author",
+      "name": "1",
```
Comment (nikhedonia) [inline packages/provider/__tests__/__snapshots__/provider-state-transition.test.js.snap]: seems like the test is still failing as the provider renders author(slug: 1) while debouncing.
This means we display author(slug: 1) for 500ms until the debouncing finishes and we get the isLoading placeholder. 
Is that what we want ?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/3221ba46bf9d7f1df5b11779e99b079814ce2dff/dextrose-test/index.html

**Pull request merged**